examples/generative/corrdiff/datasets/dataset.py

```diff
@@ -28,6 +28,45 @@
 known_datasets = {"cwb": cwb.get_zarr_dataset}
 
 
+def init_dual_dataset_from_config(
+    dataset_cfg: dict,
+    dataloader_cfg: Union[dict, None] = None,
+    batch_size: int = 1,
+    seed: int = 0,
+    train_test_split: str = True,
+) -> Tuple[base.DownscalingDataset, Iterable]:
+    """
+    A wrapper function for managing the train-test split for the CWB dataset.
+
+    Parameters:
+    - dataset_cfg (dict): Configuration for the dataset.
+    - dataloader_cfg (dict, optional): Configuration for the dataloader. Defaults to None.
+    - batch_size (int): The number of samples in each batch of data. Defaults to 1.
+    - seed (int): The random seed for dataset shuffling. Defaults to 0.
+    - train_test_split (bool): A flag to determine whether to create a validation dataset. Defaults to True.
+
+    Returns:
+    - Tuple[base.DownscalingDataset, Iterable, Optional[base.DownscalingDataset], Optional[Iterable]]: A tuple containing the training dataset and iterator, and optionally the validation dataset and iterator if train_test_split is True.
+    """
+
+    config = copy.deepcopy(dataset_cfg)
+    config.pop('train_test_split', None)
+    (dataset, dataset_iter) = init_dataset_from_config(
+        config, dataloader_cfg, batch_size=batch_size, seed=seed
+    )
+    if train_test_split:
```

Comment:
Instead of having a seperate flag for splits can we check if valid_years has values? This could also potentially push the logic all into the dataset.

Reply:
@daviddpruitt  its a bit tricky as the validation set mightn't be specific years, people might have a random sample of the data s validation and the exact details would be in the dataloader. I think I would leave the current flag but we can change this in the future